scripts/linters/pre_commit_linter.py

```diff
@@ -0,0 +1,542 @@
+        custom_linter, third_party_linter = html_linter.get_linters(
+            html_filepaths, verbose_mode_enabled=verbose_mode_enabled)
+        custom_linters.append(custom_linter)
+        third_party_linters.append(third_party_linter)
+
+        custom_linter, third_party_linter = css_linter.get_linters(
+            config_path_for_css_in_html, html_filepaths,
+            verbose_mode_enabled=verbose_mode_enabled)
+        third_party_linters.append(third_party_linter)
+
+    if css_file_extension_type:
+        custom_linter, third_party_linter = css_linter.get_linters(
+            config_path_for_oppia_css, css_filepaths,
+            verbose_mode_enabled=verbose_mode_enabled)
+        third_party_linters.append(third_party_linter)
+
+    if py_file_extension_type:
+        custom_linter, third_party_linter = python_linter.get_linters(
+            py_filepaths, verbose_mode_enabled=verbose_mode_enabled)
+        custom_linters.append(custom_linter)
+        third_party_linters.append(third_party_linter)
+
+    return custom_linters, third_party_linters
+
+
+def _get_changed_filepaths():
+    """Returns a list of modified files (both staged and unstaged)
+
+    Returns:
+        a list of filepaths of modified files.
+    """
+    unstaged_files = subprocess.check_output([
+        'git', 'diff', '--name-only',
+        '--diff-filter=ACM']).splitlines()
+    staged_files = subprocess.check_output([
+        'git', 'diff', '--cached', '--name-only',
+        '--diff-filter=ACM']).splitlines()
+    all_changed_filepaths = unstaged_files + staged_files
+    return [filepath for filepath in all_changed_filepaths]
+
+
+def _get_all_files_in_directory(dir_path, excluded_glob_patterns):
+    """Recursively collects all files in directory and
+    subdirectories of specified path.
+
+    Args:
+        dir_path: str. Path to the folder to be linted.
+        excluded_glob_patterns: set(str). Set of all glob patterns
+            to be excluded.
+
+    Returns:
+        a list of files in directory and subdirectories without excluded files.
+    """
+    files_in_directory = []
+    for _dir, _, files in os.walk(dir_path):
+        for file_name in files:
+            filepath = os.path.relpath(
+                os.path.join(_dir, file_name), os.getcwd())
+            if not any([fnmatch.fnmatch(filepath, gp) for gp in
+                        excluded_glob_patterns]):
+                files_in_directory.append(filepath)
+    return files_in_directory
+
+
+def _get_file_extensions(file_extensions_to_lint):
+    """This function is used to return the file extensions which need to be
+    linted and checked.
+
+    Args:
+        file_extensions_to_lint: list(str). The list of file extensions to be
+        linted checked.
+
+    Returns:
+        all_file_extensions_type: list(str). The list of all file extensions
+        to be linted and checked.
+    """
+    all_file_extensions_type = ['js', 'py', 'html', 'css', 'other']
```

Comment:
@Hudda, Why `ts` is not added here?

Reply:
@DubeySandeep we do not have separate linter for js and ts, if both options used together then Js/Ts linter will run twice.

Reply:
So how the general-purpose check would run for ts files?

Reply:
@DubeySandeep Thanks for pointing this out. I will open a PR to fix this issue.

Reply:
Can you please resolve it asap! (We have an open issue for the same at #8993 and I've assigned you to the same.)

(Note: This is a critical issue, a PR can be merged with `fit` and this can cause CI to run only one test.)